timemachine/cpp/src/kernels/kernel_utils.cuh

```diff
@@ -31,7 +31,12 @@
 
 #define ONE_4PI_EPS0 138.935456
 
+// default_threads_per_block should be at least 128 to ensure maximum occupancy for Cuda Arch 8.6 with 48 SMs
+// given that there aren't too many registers in the kernel.
+// Refer to the occupancy calculator in Nsight Compute for more details
+static const int default_threads_per_block = 128;
 static const int warp_size = 32;
+static_assert(default_threads_per_block % warp_size == 0); // default_threads_per_block must be multiple of warp_size
```

Comment:
Nice -- I remember being thrown off by this in my first TM code review https://github.com/proteneer/timemachine/pull/278#discussion_r501403872

Reply:
I should actually clarify this comment as 'must' is imprecise. It simply that it is wasteful for the threads per block not being a multiple of warp size

Reply:
Updated here https://github.com/proteneer/timemachine/pull/1047/commits/e946fdbc5d44d9dbf53f44be1eb7715245e05a96